### PR TITLE
Creating use case tests for task selection [WIP]

### PR DESCRIPTION
**Pull request by thiagoyeds**

### CODE REVIEW

File: src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java [hunk 1]
```diff
@@ -1,176 +1,319 @@
...
+		properties.put(SapsPropertiesConstants.IMAGE_DATASTORE_PORT, "");
+		properties.put(SapsPropertiesConstants.IMAGE_WORKER, "");
+		properties.put(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_SUBMISSOR, "");
+		properties.put(SapsPropertiesConstants.SAPS_EXECUTION_PERIOD_CHECKER, "");
+		properties.put(SapsPropertiesConstants.ARREBOL_BASE_URL, "");
+
+		try {
+			return new Scheduler(properties, imageStore, null, arrebol, selector);
+		} catch (SapsException | SQLException e) {
+			e.printStackTrace();
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java]: Please, do not catch an exception just to print the stack trace, just throw it.

File: src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java [hunk 2]
```diff
@@ -1,176 +1,319 @@
...
 	
 	@Test
-	public void testMapUserToTasksWithTasksL1Created() {
-		Map<String, List<ImageTask>> mapUserToTasks = scheduler.mapUsers2Tasks(tasksL1Created);
-		
-		assert (mapUserToTasks.get("user1").size() == 3);
-		assert (mapUserToTasks.get("user2").size() == 2);
-		assert (mapUserToTasks.get("user3").size() == 1);
-		
+	public void testSelectZeroTasksWithZeroCapacitySubmitionWhenThereIsNoAvailableTasks()
```
Comment (paulofelipefeitosa) [inline src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java]: _Submition_ is a typo - **Submission**

**Review by paulofelipefeitosa — CHANGES_REQUESTED** [review]
Hi @ThiagoWhispher thank you for the test set, I think that you get it :smile:! Now its time to improve the test code! 

I noticed that there is a lot of duplicated code, in truth the test logic is the same for all test cases: you will provide a test data and then check if the returned value is equal to the expected value, correct? So, I think that we could have just one test code and reuse it with different parameters and expected values, please check [JUnit Parameterized Tests](https://github.com/junit-team/junit4/wiki/parameterized-tests).

File: src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java (hunk 2 above)
Comment (thiagoyeds) [reply src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java]: Done

File: src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java (hunk 1 above)
Comment (thiagoyeds) [reply src/test/java/org/fogbowcloud/saps/engine/core/scheduler/SchedulerTest.java]: Done

**Comment by thiagoyeds** [comment]
Thank you @paulofelipefeitosa, I will see your recommendation using JUnit Parameterized Tests and apply to tests.

**Comment by thiagoyeds** [comment]
This PR continues in PR #53

**Pull request closed without merging**